dll/win32/comctl32/propsheet.c

```diff
@@ -3580,6 +3580,27 @@ PROPSHEET_DialogProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
         }
         SetFocus(GetDlgItem(hwnd, IDOK));
       }
+#ifdef __REACTOS__
```

Comment:
Out of curiosity, what is the computed initial size of the dialog and its position when it is created, before running the added code?

Reply:
In the file "win32ss / user / user32 / windows / dialog.c"
In the DIALOG_CreateIndirect (...) function (line 898), before calling CreateWindowEx (...) the code attempts to adjust the position of the window within the screen area if possible.
For this calculates:
dY = pos.y + size.cy - rcWork.bottom
if (dY> 0) pos.y - = dY;
this size.cy was read from IDD_PROPSHEET TEMPLATE structure
and has a value of 140 (parent window).
Here we need the correct window height value which is that of the IDD_TASKBARPROP_TASKBAR TEMPLATE and is equal to (218 + WIZARD_PADDING * 3 + Default_Buttons_height) to subtract from the current position which is the position of the top left corner of the start button.

Reply:
Can you add a brief comment describing this to the top of the code here? Without context, this looks like a hack, but your description makes perfect sense.
E.g. "user32 positions the dialog based on the IDD_PROPSHEET template, but we've since made it larger by adding controls"